### PR TITLE
Add per-model cost breakdown to Slack job notifications

### PR DESCRIPTION
**Pull request by ashesfall**
Add JobStatsStore.formatModelCostLines() for multi-line Slack output showing one cost entry per model with provider prefix stripped (e.g. "openrouter/" removed). Update SlackNotifier to use the new format and show runner totals in brackets on the total cost line. Add tests covering provider prefix stripping, zero-cost omission, and multi-segment model keys like "openrouter/minimax/minimax-m2.7".

### CODE REVIEW

File: flowtree/runtime/src/main/java/io/flowtree/slack/SlackNotifier.java [hunk 1]
```diff
@@ -1152,10 +1152,21 @@ private void appendSessionMetrics(StringBuilder sb, JobCompletionEvent event) {
 
         // Cost
         if (event.getCostUsd() > 0) {
-            sb.append("   :moneybag: Cost: $").append(String.format("%.2f", event.getCostUsd()));
-            sb.append(JobStatsStore.formatCostBreakdown(event.getCostByRunner()));
-            sb.append(JobStatsStore.formatCostBreakdown(event.getCostByModel()));
-            sb.append("\n");
+            sb.append(JobStatsStore.formatModelCostLines(event.getCostByModel()));
+            sb.append("   :dollar: $").append(String.format("%.2f", event.getCostUsd()));
+            // TODO(review): if costByRunner is null/empty or all values are zero, output is "total []" with empty brackets
```
[the remark this answers is not in the document]
Comment (ashesfall) [reply flowtree/runtime/src/main/java/io/flowtree/slack/SlackNotifier.java]: Removed the `TODO(review)` marker. The empty brackets issue is now handled by conditionally appending the runner segment only when there are non-zero entries — see the fix for comment 3319549256.

File: flowtree/runtime/src/main/java/io/flowtree/slack/SlackNotifier.java [hunk 2]
```diff
@@ -1152,10 +1152,21 @@ private void appendSessionMetrics(StringBuilder sb, JobCompletionEvent event) {
...
+            if (costByRunner != null && !costByRunner.isEmpty()) {
+                boolean first = true;
+                for (Map.Entry<String, Double> entry : costByRunner.entrySet()) {
+                    if (entry.getValue() == null || entry.getValue() <= 0.0) continue;
+                    if (!first) sb.append(" | ");
+                    first = false;
+                    sb.append(entry.getKey()).append(" $").append(String.format("%.2f", entry.getValue()));
+                }
+            }
+            sb.append("]\n");
```
[the remark this answers is not in the document]
Comment (ashesfall) [reply flowtree/runtime/src/main/java/io/flowtree/slack/SlackNotifier.java]: Fixed by restructuring the cost block logic to build the runner segment first and only append it when there are non-zero entries. The change at `SlackNotifier.java:1153-1172`:

- Before: unconditionally opened ` total [` and closed with `]`, producing `total []` when all runners were zero/null/empty
- After: builds runner content into a StringBuilder first, and only appends ` [runnerContent]` when runnerSb.length() > 0. When costByRunner is null, empty, or contains only zero values, the output is now just `:dollar: $X.XX total` without brackets.

The new test `testNotifierCostBlockWithRunnerBreakdown` verifies all three cases: (1) cost with non-empty runner breakdown, (2) cost with empty costByRunner (no brackets), (3) cost with only zero values in costByRunner (no brackets).

[the remark this answers is not in the document]
Comment (ashesfall) [reply flowtree/runtime/src/main/java/io/flowtree/slack/SlackNotifier.java]: Added `testNotifierCostBlockWithRunnerBreakdown` which tests the cost block via `setMessageCallback` following the same convention as `testNotifierMessageFormatting`. The test covers three cases:

1. **Cost with both costByModel and costByRunner** — verifies `:dollar: $1.75 total [claude $1.20 | opencode $0.55]` and model cost lines are present
2. **Cost with empty costByRunner** — verifies `:dollar: $0.80 total` appears without brackets (edge case from the TODO)
3. **Cost with only zero values in costByRunner** — verifies `:dollar: $0.10 total` without brackets when all runner entries are 0.0

The test is in `SlackIntegrationTest.java:177` and uses the same `setMessageCallback`/JSON-extraction pattern as the existing tests.
